internal/types/document_validation.go

```diff
@@ -79,5 +79,14 @@ func (d *Document) ValidateData() error {
 		return newValidationError(fmt.Errorf("invalid document: document must contain '_id' field"))
 	}
 
+	v := must.NotFail(d.Get("_id"))
+
+	switch v.(type) {
+	case Array, *Array:
+		return fmt.Errorf("The '_id' value cannot be of type array")
+	case Regex, *Regex:
```

Comment:
```suggestion
	case Regex:
```

Scalar value should never be passed by pointer